post-processing/post_processing.py

```diff
@@ -1,41 +1,179 @@
+import argparse
+import errno
 import fileinput
+from functools import reduce
+import operator as op
+import os
 import pandas as pd
 import re
+import traceback
+import yaml
 
-def get_display_name_params(display_name):
-    """
-        Return a dictionary of parameter names and their values from the given input string, if present.
+class PostProcessing:
 
-        Args:
-            display_name: str, expecting a format of <test_name> followed by zero or more %<param>=<value> pairs.
+    def __init__(self, debug=False, verbose=False):
+        self.debug = debug
+        self.verbose = verbose
+
+    def run_post_processing(self, log_path, config):
+        """
+            Return a dataframe containing the information passed to a plotting script and produce relevant graphs.
+
+            Args:
+                log_path: str, path to a log file or a directory containing log files.
+                config: dict, configuration information for plotting.
+        """
+
+        log_files = []
+        # look for perflogs
+        if os.path.isfile(log_path):
+            log_files = [log_path]
+        elif os.path.isdir(log_path):
+            log_files = [os.path.join(root, file) for root, _, files in os.walk(log_path) for file in files]
+        else:
+            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), log_path)
+
+        if self.debug:
+            print("Found log files:")
+            for log in log_files:
+                print("-", log)
+            print("")
+
+        df = pd.DataFrame()
+        # put all perflog information in one dataframe
+        for file in log_files:
+            try:
+                temp = read_perflog(file)
+                df = pd.concat([df, temp], ignore_index=True)
+            except KeyError as e:
+                if self.debug:
+                    print("Discarding %s:" %os.path.basename(file), type(e).__name__ + ":", e.args[0], e.args[1])
+                    print("")
+        if df.empty:
+            raise FileNotFoundError(errno.ENOENT, "Could not find a valid perflog in path", log_path)
+
+        columns = config["columns"]
+        REQUIRED_COLUMNS = [r"\w+_value$", r"\w+_unit$"]
+        required_column_matches = [len(list(filter(re.compile(rexpr).match, columns))) > 0 for rexpr in REQUIRED_COLUMNS]
+        # check for required columns
+        if (len(columns) < 3) | (False in required_column_matches):
+            raise KeyError("Config must contain at least 3 specified columns: a figure of merit value, a figure of merit unit, and something to plot the figure of merit against", columns)
+
+        invalid_columns = []
+        # check for invalid columns
+        for col in columns:
+            if col not in df.columns:
+                invalid_columns.append(col)
+        if invalid_columns:
+            raise KeyError("Could not find columns", invalid_columns)
+
+        filters = config["filters"]

```

Comment:
A check just like the one you have above, to make sure the needed columns for the filters are included in the dataframe, has to be added here. Same for the datasets.
Note that this check has to be here and not in the `read_config`, as is compares to the df.